src/components/select/select.js

```diff
@@ -202,13 +203,19 @@ function SelectDirective($mdSelect, $mdUtil, $mdTheming, $mdAria, $compile, $par
       var valueEl = element.find('md-select-value');
       var isReadonly = angular.isDefined(attr.readonly);
 
+      mdSelectCtrl.mdSelectHeaderInput = element.find('md-select-header').find('input')[0];
```

Comment:
Ahh yes, this is reasonable. 

I moved the check to be closer to the usage of it. Previously it was used in another check (for handling `keydown` stuff. I put it on the `mdSelectCtrl` because that's the only controller that was present both here and in the `onMenuKeyDown` function. 

But that code is gone now, so all good!